### PR TITLE
modbus: Document exception codes

### PR DESCRIPTION
**Pull request by kartben**
Add Doxygen comments for modbus exception codes.

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

**Review by aasinclair — DISMISSED** [review]

File: include/zephyr/modbus/modbus.h [hunk 1]
```diff
@@ -45,16 +45,16 @@ extern "C" {
...
+#define MODBUS_EXC_NONE				0    /**< No exception */
+#define MODBUS_EXC_ILLEGAL_FC			1    /**< Illegal function code */
+#define MODBUS_EXC_ILLEGAL_DATA_ADDR		2    /**< Illegal data address */
+#define MODBUS_EXC_ILLEGAL_DATA_VAL		3    /**< Illegal data value */
+#define MODBUS_EXC_SERVER_DEVICE_FAILURE	4    /**< Server device failure */
+#define MODBUS_EXC_ACK				5    /**< Acknowledge */
+#define MODBUS_EXC_SERVER_DEVICE_BUSY		6    /**< Server device busy */
+#define MODBUS_EXC_MEM_PARITY_ERROR		8    /**< Memory parity error */
+#define MODBUS_EXC_GW_PATH_UNAVAILABLE		10   /**< Gateway path unavailable */
+#define MODBUS_EXC_GW_TARGET_FAILED_TO_RESP	11   /**< Gateway target device failed to respond */
```
Comment (jfischer-no) [inline include/zephyr/modbus/modbus.h:57]: The length of this line is at the limit, this style does not scale, please find another way (I have already asked you) to document this.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: include/zephyr/modbus/modbus.h (hunk 1 above)
Comment (kartben) [reply include/zephyr/modbus/modbus.h]: updated, thx!

Comment (jfischer-no) [reply include/zephyr/modbus/modbus.h]: @kartben Thank you. There are now two typos, `#define MODBUS_EXC_GW_TARGET_FAILED_TO_RESP	1`..., please recheck.

Comment (kartben) [reply include/zephyr/modbus/modbus.h]: That was really bad, sorry!

**Review by jfischer-no — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]

**Pull request merged**